File: Parsing/AST/FileAstNode.cs
using LLVMSharp;

namespace Raccoonlang.Parsing.AST;

using Tokenizing;

public class FileAstNode
{
    public ImportsAstNode Imports { get; set; }
    public NamespaceAstNode? NameSpace { get; set; }
    public TypeDefinitionsAstNode Typedefs { get; set; }

    public static FileAstNode Parse(Parser parser)
    {
        FileAstNode node = new FileAstNode();

        node.NameSpace = NamespaceAstNode.TryParse(parser);
        node.Imports = ImportsAstNode.Parse(parser);
        node.Typedefs = TypeDefinitionsAstNode.Parse(parser);

        return node;
    }

    public override string ToString()
    {
        return "FileAstNode{"+
                "imports="+ Imports +
                ", nameSpace=[" + NameSpace +
                "], typedefs=["+ string.Join(", ", Typedefs.NodeList) +
                "]}";
    }

    public void Compile(LLVMContextRef context)
    {
        
    }
}

File: src/main/java/raccoonlang/parser/ast/FileAstNode.java
package raccoonlang.parser.ast;

import raccoonlang.parser.Parser;
import raccoonlang.tokenizer.TokenStream;

public class FileAstNode {
    public ImportsAstNode importsAstNode;
    public NamespaceAstNode namespaceAstNode;
    public TypeDefinitionsAstNode typeDefinitionsAstNode;

    @Override
    public String toString() {
        return "FileAstNode{" +
                "importsAstNode=" + importsAstNode.toString() +
                ", namespaceAstNode=" + (namespaceAstNode == null ? "" : namespaceAstNode.toString()) +
                //", typeDefinitionsAstNode=" + typeDefinitionsAstNode.toString() +
                '}';
    }

    public static FileAstNode parse(Parser parser){
        FileAstNode fileAstNode = new FileAstNode();

        fileAstNode.importsAstNode = ImportsAstNode.parse(parser);
        fileAstNode.namespaceAstNode = NamespaceAstNode.tryParse(parser);

        return fileAstNode;
    }
}
